File: backend/databasehandler/src/main/java/com/pixtranslator/backend/databasehandler/TranslateddictionaryRepository.java
package com.pixtranslator.backend.databasehandler;

import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

import java.io.Serializable;
import java.util.Collection;
import java.util.List;

@RepositoryRestResource(collectionResourceRel = "dict", path = "dict")
public interface TranslateddictionaryRepository extends CrudRepository<Translateddictionary, Serializable> {

  Translateddictionary findByGerman(@Param("german") String german);

  List<Translateddictionary> findAllByGermanInIgnoreCase(@Param("german") Collection<String> german);
}

File: backend/databasehandler/src/main/kotlin/com/pixtranslator/backend/databasehandler/TranslateddictionaryRepository.kt
package com.pixtranslator.backend.databasehandler

import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import java.util.Collection
import java.util.List

@RepositoryRestResource(collectionResourceRel = "dict", path = "dict")
interface TranslateddictionaryRepository : CrudRepository<Translateddictionary, Long> {
  fun findByGerman(@Param("german") german: String): Translateddictionary?
  fun findAllByGermanInIgnoreCase(@Param("german") german: Collection<String>): List<Translateddictionary>
}